packages/terra-popup/src/_PopupContent.jsx

```diff
@@ -123,21 +77,37 @@ const propTypes = {
 };
 
 const defaultProps = {
-  arrowPosition: 'top',
   classNameInner: null,
-  closeOnEsc: false,
-  closeOnOutsideClick: false,
-  closeOnResize: false,
   contentHeightMax: -1,
   contentWidthMax: -1,
   isHeaderDisabled: false,
+  isHeightDynamic: false,
+  isWidthDynamic: false,
 };
 
 class PopupContent extends React.Component {
-  static getContentStyle(arrow, arrowPosition, height, maxHeight, width, maxWidth) {
-    const validHeight = maxHeight <= 0 || height <= maxHeight ? height : maxHeight;
-    const validWidth = maxWidth <= 0 || width <= maxWidth ? width : maxWidth;
-    return { height: `${validHeight.toString()}px`, width: `${validWidth.toString()}px` };
+  static getDimensionStyle(value, maxValue, isDynamic) {
+    if (value > 0) {
```

Comment:
Could we use a guard clause here? 
```
if (value <= 0) {
  return null;
}
....
```

Reply:
The return null also acts as an "else" if we aren't constrained and we are dynamic to allow for the initial layout.